internal/unix/types_other.go

```diff
@@ -0,0 +1,57 @@
+// +build !linux
+
+package unix
+
+import (
+	"fmt"
+	"runtime"
+)
+
+var errNonLinux = fmt.Errorf("unsupported platform %s/%s", runtime.GOOS, runtime.GOARCH)
+
+const (
+	ENOENT           = 0x2
+	BPF_OBJ_NAME_LEN = 0x10
+	BPF_TAG_SIZE     = 0x8
+	SYS_BPF          = 321
+	EAGAIN           = 0xb
+	F_DUPFD_CLOEXEC  = 0x406
+	ENOSPC           = 0x1c
+	EINVAL           = 0x16
+)
+
+// Statfs_t is a wrapper
+type Statfs_t struct {
+	Type    int64
+	Bsize   int64
+	Blocks  uint64
+	Bfree   uint64
+	Bavail  uint64
+	Files   uint64
+	Ffree   uint64
+	Fsid    [2]int32
+	Namelen int64
+	Frsize  int64
+	Flags   int64
+	Spare   [4]int64
+}
+
+// Syscall is a wrapper
+func Syscall(trap, a1, a2, a3 uintptr) (r1, r2, err uintptr) {
+	return 0, 0, 0
```

Comment:
Let's make sure this returns an error:

```suggestion
	return 0, 0, -1
```

Reply:
i've tried this, but get the following error:
`constant -1 overflows uintptr`

Reply:
is it ok, to return `1` instead?
according to `include/asm/errno.h` `1` indicates `Operation not permitted`.

Reply:
i've updated the PR and return now `syscall.Errno(1)`.

Reply:
Yeah, that sounds good. I was looking at `man syscall` which says it will return -1, sorry.